src/graphics/graphics-device.js

```diff
@@ -822,6 +824,16 @@ class GraphicsDevice extends EventHandler {
         this.gl = null;
     }
 
+    // provide webgl implementation for the vertex buffer
+    implementVertexBuffer(vertexBuffer) {
```

Comment:
Not a big deal since we can change it any time, but I find these names confusing.

How about `createVertexBufferImpl`?

Reply:
done .. I was originally considering those names too